### PR TITLE
Improvements and fix #9

### LINKED ISSUE #9: "this.sync.localforageKey" doesn't get updated
I noticed a strange behavior in the `sync()` method when using *localFrage-backbone*. It only set the `this.sync.localforageKey` one time. After creating and fetching another model, it never updated this key. 

To fix this I added on line in the `sync` method:
```
this.sync.localforageKey = name + "/" + model.id

```

[Add after line 67 in localforage.backbone.js](https://github.com/mozilla/localForage-backbone/blob/master/src/localforage.backbone.js#L67)

Did anyone noticed the same issue?
After adding this it works for me.

Pull request?

### PR DESCRIPTION
**Pull request by stephanebachelier**
Fix #9 

A better solution than my previous work in #13.

* improve model.destroy test to add an expectation which was not the case
* upgrade grunt deps
* add jscs 
* lint source files with jshint and jscs (same as localforage)

### CODE REVIEW

File: test/backbone.model.js [hunk 1]
```diff
@@ -63,12 +63,26 @@ describe('Backbone.Model', function () {
...
-                      error: function () {
-                        done();
+                success: function(model, resp, options) { // jshint unused:false
+                    expect(model.attributes).toEqual(resp);
+                    var handlers = {
+                      success: function() {
+                        testComplete();
+                      },
+                      error: function() {
+                        testComplete();
```
Comment (tofumatt) [inline test/backbone.model.js]: These all look to be two spaces when the linter checks for 4, no?

**Comment by tofumatt** [comment]
You should be able to add jasmine as an npm package, instead of checking in all the code into vendor. That would reduce the code checked in by this commit by a lot :smile:

**Comment by stephanebachelier** [comment]
@tofumatt you're right for jasmine

File: test/backbone.model.js (hunk 1 above)
Comment (stephanebachelier) [inline test/backbone.model.js]: @tofumatt I need to check why the linter failed to catch this.

Comment (stephanebachelier) [inline test/backbone.model.js]: @tofumatt there was no configuration enforcing a 4 spaces indentation. I've sent you a PR in localForage as well.

**Comment by stephanebachelier** [comment]
@tofumatt update PR as requested. 
By the way lint tools where not enforcing 4 spaces indentation.

**Comment by tofumatt** [comment]
Looks great, thanks! :+1: 

I'll get to publishing a new release in a bit. :-)

**Pull request merged**